src/receive_imf.rs

```diff
@@ -372,25 +372,52 @@ async fn get_to_and_past_contact_ids(
             }
         }
         ChatAssignment::AdHocGroup => {
-            to_ids = add_or_lookup_contacts_by_address_list(
-                context,
-                &mime_parser.recipients,
-                if !mime_parser.incoming {
-                    Origin::OutgoingTo
-                } else if incoming_origin.is_known() {
-                    Origin::IncomingTo
-                } else {
-                    Origin::IncomingUnknownTo
-                },
-            )
-            .await?;
+            let key_to_ids = match mime_parser.was_encrypted() {
+                false => Vec::new(),
+                true => {
+                    let ids = lookup_key_contacts_by_address_list(
+                        context,
+                        &mime_parser.recipients,
+                        to_member_fingerprints,
+                        None,
+                    )
+                    .await?;
+                    match ids.contains(&None) {
```

Comment:
This reflects the old `should_encrypt()` behavior: if we don't have any peerstate, we don't want to encrypt.